src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEFileSaveWatcher.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.VS.Automation;
+using VSLangProj;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    /// <summary>
+    /// This class handles being notified whenever a source file is saved by the hierarchy
+    /// </summary>
+    [Export(typeof(IFileActionHandler))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
```

Comment:
We don't need this in shared projects; this should be CSharpOrVisualBasicLanguageServices.

Reply:
We had this conversation from last PR that wasn't resolved: https://github.com/dotnet/project-system/pull/4318#discussion_r236899719.

Reply:
There were a few questions I asked on that PR that didn't get answered, but the commit stream was getting too noisy so I closed it. Now that I've finished all of the functionality I intend this to be the "real" PR so will keep things cleaner, and hopefully will get some more answers. That comment thread you linked to still doesn't clarify for me what you mean by suggesting I change the `AppliedTo` as that capability doesn't exist.

Reply:
CSharpOrVisualBasic is not correct, given this is not applicable to Shared Projects.
DotNetLanguageService is not correct given this is not applicable to F#.

This needs to be either "CSharpOrVisualBasic" + "LanguageService", a new capability.